From: mscharged-port
Subject: Substitute strlen+itoa with snprintf
Upstream-base: a353c8f2dfb34eb08fecd725965a46ea629428e3

Changed strlen+itoa to a safe snprintf. itoa was only present in MWCC's libc
and isn't standard C/C++.

diff --git a/src/RVL_SDK/dwc/dwc_encsession.c b/src/RVL_SDK/dwc/dwc_encsession.c
index 9020173..65b7601 100644
--- a/src/RVL_SDK/dwc/dwc_encsession.c
+++ b/src/RVL_SDK/dwc/dwc_encsession.c
@@ -14,7 +14,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+#if !defined(MSCHARGED_NATIVE)
 #include <extras.h>
+#endif
 
 const char* DWCi_ENC_URL_BASE = NULL;
 static int s_local_seed;
@@ -574,7 +576,7 @@ DWCiRankingSessionResult DWCi_EncSessionGetAsync(const char* url,
             + strlen(g_session.gamename)
             + strlen(url)
             + strlen("?pid=")
-#if defined(VERSION_R4QK01)
+#if defined(VERSION_R4QK01) || defined(MSCHARGED_NATIVE)
             + snprintf(buf, sizeof(buf), "%d", pid)
 #else
             + strlen(itoa(pid, buf, 10))
